### PR TITLE
ENH: Use minimum system zstd in Linux scripts

### PR DESCRIPTION
**Pull request by tbirdso**
Changes:
- Add `--long=31` option for improved Linux build archive compression
- Verify that zstd is at least v1.3.2 where `--long` is introduced
- Remove zstd v1.2.0 download option

Required for v5.3rc04.post4 builds which add Python 3.11 archives. Without the `--long=31` option these archives are over the 2GB limit imposed by Github. With the `--long=31` option build archives are right at the threshold at about 1.97GB. https://github.com/InsightSoftwareConsortium/ITKPythonBuilds/releases/tag/v5.3rc04.post4

I have updated zstd on `blaster` via `sudo apt install zstd` such that the installed zstd/unzstd version is now v1.4.4. This PR introduces the requirement that `unzstd>1.3.2` is used to decompress archives for building module wheels on Linux.

Echoes compression options introduced in https://github.com/InsightSoftwareConsortium/ITKPythonPackage/commit/de54e148584827cc3cb89898358510877762f1b7 that were reverted due to zstd compatibility on Linux at the time.

I have tested by
1. Running `dockcross-manylinux-build-tarball.sh` to build archives on `blaster`;
2. Running `dockcross-manylinux-download-cache-and-build-module-wheels.sh` to build ITKSplitComponents module wheels in CI: https://github.com/InsightSoftwareConsortium/ITKSplitComponents/actions/runs/3490135651/jobs/5841170399

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]
Code looks good, I have not tried running it.

**Comment by tbirdso** [comment]
Thanks @dzenanz . I've updated the description with notes on testing.

**Comment by dzenanz** [comment]
Feel free to merge if you don't want to wait for Matt's review.

**Comment by tbirdso** [comment]
@dzenanz  sounds good, will wait for all Linux Python jobs in https://github.com/InsightSoftwareConsortium/ITKSplitComponents/actions/runs/3490135651/jobs/5841171620 to complete before merge

**Review by thewtex — APPROVED** [review]
:+1:

**Comment by tbirdso** [comment]
Python builds in ITKSplitComponents are passing 👍

**Pull request merged**
